### PR TITLE
BaSP-TG-82: Task redux - Get

### PR DESCRIPTION
**Pull request by renzopimentel**
Create constants to GET tasks
Create actions to GET tasks
Create thunk to GET tasks
Create tasks reducer
Connect list component with store to GET tasks
Implement spinner while fetch is pending
modify modalMessage to show correct error

### CODE REVIEW

File: src/redux/tasks/reducer.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  modalContent: { title: '', content: '' },
+  showModalMessage: false
+};
+
+const tasksReducer = (state = INITIAL_STATE, action) => {
+  switch (action.type) {
+    case GET_TASKS_PENDING: {
+      return {
+        ...state,
+        isLoading: false
```
Comment (CristianBucci) [inline src/redux/tasks/reducer.js:21]: In this case, isLoading should be true!

File: src/redux/tasks/reducer.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+      return {
+        ...state,
+        list: action.payload,
+        isLoading: false
+      };
+    }
+    case GET_TASKS_ERROR: {
+      return {
+        ...state,
+        list: [],
```
Comment (CristianBucci) [inline src/redux/tasks/reducer.js:34]: There is no need to add  list here!

File: src/Components/Tasks/index.js [hunk 3]
```diff
@@ -1,17 +1,20 @@
...
 
 const Tasks = () => {
-  const [tasksList, setTasksList] = useState([]);
   const [showModalConfirm, setShowModalConfirm] = useState(false);
   const [showModalMessage, setShowModalMessage] = useState(false);
   const [modalContent, setModalContent] = useState({ title: 'title', content: 'content' });
   const [itemId, setItemId] = useState(null);
   const location = useLocation();
+  const dispatch = useDispatch();
+  const tasksList = useSelector((state) => state.tasks.list);
```
Comment (CristianBucci) [inline src/Components/Tasks/index.js:18]: You need to use the isLoading to display the spinner while the action is pending! you can do it like this:
const {list: tasksList,
           isLoading} = useSelector((state) => state.tasks);

File: src/Components/Tasks/index.js [hunk 4]
```diff
@@ -26,20 +29,12 @@ const Tasks = () => {
...
-      setTasksList(data.data);
-    } catch (error) {
-      setModalContent({ title: 'ERROR!', content: `Could not GET Tasks! ${error.message}` });
-      setShowModalMessage(true);
-    }
+  useEffect(() => {
+    dispatch(getTasks());
   }, []);
 
   return (
```
Comment (CristianBucci) [inline src/Components/Tasks/index.js:40]: You should display the spinner while isLoading is true, you can check th PR #21

File: src/Components/Tasks/index.js [hunk 5]
```diff
@@ -26,20 +29,12 @@ const Tasks = () => {
...
-      const response = await fetch(`${process.env.REACT_APP_API_URL}/tasks`);
-      const data = await response.json();
-      setTasksList(data.data);
-    } catch (error) {
-      setModalContent({ title: 'ERROR!', content: `Could not GET Tasks! ${error.message}` });
-      setShowModalMessage(true);
-    }
+  useEffect(() => {
+    dispatch(getTasks());
   }, []);
```
Comment (ignacioaristo) [inline src/Components/Tasks/index.js:38]: You can add the taskList in order to re render and show the updated list.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/redux/tasks/reducer.js (hunk 1 above)
Comment (renzopimentel) [reply src/redux/tasks/reducer.js:21]: Perfect! Changes has been updated

**Review by nicolasblasco — APPROVED** [review]

**Review by CristianBucci — APPROVED** [review]

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**